packages/node_modules/@node-red/nodes/core/sequence/19-batch.js

```diff
@@ -132,32 +133,38 @@ module.exports = function(RED) {
             gids.push(gid);
         }
         var group = groups[gid];
-        group.msgs.push(msg);
+        group.msgs.push(msgInfo);
         if ((group.count === undefined) &&
-            msg.parts.hasOwnProperty('count')) {
-            group.count = msg.parts.count;
+            msgInfo.msg.parts.hasOwnProperty('count')) {
+            group.count = msgInfo.msg.parts.count;
         }
     }
 
-    function concat_msg(node, msg) {
+    function concat_msg(node, msg, send, done) {
         var topic = msg.topic;
         if(node.topics.indexOf(topic) >= 0) {
             if (!msg.hasOwnProperty("parts") ||
                 !msg.parts.hasOwnProperty("id") ||
                 !msg.parts.hasOwnProperty("index") ||
                 !msg.parts.hasOwnProperty("count")) {
-                node.error(RED._("batch.no-parts"), msg);
+                done(RED._("batch.no-parts"));
                 return;
             }
             var gid = msg.parts.id;
             var pending = node.pending;
-            add_to_topic_group(pending, topic, gid, msg);
+            add_to_topic_group(pending, topic, gid, {msg, send, done});
             node.pending_count++;
             var max_msgs = max_kept_msgs_count(node);
             if ((max_msgs > 0) && (node.pending_count > max_msgs)) {
+                Object.values(node.pending).forEach(p_topic => {
+                    Object.values(p_topic.groups).forEach(group => {
+                        group.msgs.forEach(msg => {
+                            msg.done(RED._("batch.too-many"))
```

Comment:
As with other comments - is it really the right thing to throw an error against every message in the group? That is going to cause a lot of noise in the log.

Reply:
as commented in `split.js`(https://github.com/node-red/node-red/pull/2402#discussion_r433017384),  I propose following behaviour:
- call done() for queued messages,
- call done(RED._("batch.too-many")) for the message which triggers overflow, and
- call node.trace() to log the flashed messages

Reply:
I agree with calling `done()` on the queued messages and `done(RED._("batch.too-many"))` on the message that caused the overflow.

I do not think we should `node.trace` all of the flushed messages as again, that could be a huge list and there's little the user would be able to do with that information from the trace log.

Reply:
I agree with this.  I'll replace `done(RED._("batch.too-many"))` with `done()`, and emit a single error on the message that caused the overflow.